setup.py

```diff
@@ -155,35 +90,33 @@ def gen_packages_items():
     return packages
 
 
-if __name__ == '__main__':
-    write_version_py()
-    setup(
-        name='mmcls',
-        version=get_version(),
-        description='OpenMMLab Image Classification Toolbox and Benchmark',
-        long_description=readme(),
-        author='OpenMMLab',
-        author_email='yangleidev@gmail.com',
-        keywords='computer vision, image classification',
-        url='https://github.com/open-mmlab/mmclassification',
-        packages=find_packages(exclude=('configs', 'tools', 'demo')),
-        package_data={'mmcls.ops': ['*/*.so']},
-        classifiers=[
-            'Development Status :: 4 - Beta',
-            'License :: OSI Approved :: Apache Software License',
-            'Operating System :: OS Independent',
-            'Programming Language :: Python :: 3',
-            'Programming Language :: Python :: 3.5',
-            'Programming Language :: Python :: 3.6',
-            'Programming Language :: Python :: 3.7',
-        ],
-        license='Apache License 2.0',
-        setup_requires=parse_requirements('requirements/build.txt'),
-        tests_require=parse_requirements('requirements/tests.txt'),
-        install_requires=parse_requirements('requirements/runtime.txt'),
-        extras_require={
-            'all': parse_requirements('requirements.txt'),
-            'tests': parse_requirements('requirements/tests.txt'),
-            'build': parse_requirements('requirements/build.txt'),
-        },
-        zip_safe=False)
+setup(
+    name='mmcls',
+    version=get_version(),
+    description='OpenMMLab Image Classification Toolbox and Benchmark',
+    long_description=readme(),
+    author='OpenMMLab',
+    author_email='yangleidev@gmail.com',
+    keywords='computer vision, image classification',
+    url='https://github.com/open-mmlab/mmclassification',
+    packages=find_packages(exclude=('configs', 'tools', 'demo')),
+    package_data={'mmcls.ops': ['*/*.so']},
+    classifiers=[
+        'Development Status :: 4 - Beta',
+        'License :: OSI Approved :: Apache Software License',
+        'Operating System :: OS Independent',
+        'Programming Language :: Python :: 3',
+        'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.6',
+        'Programming Language :: Python :: 3.7',
+    ],
+    license='Apache License 2.0',
+    setup_requires=parse_requirements('requirements/build.txt'),
```

Comment:
Actually `build.txt` is unnecessary.

Reply:
so ``setup_requires='['pytest-runner']``?

Reply:
We may just remove this line and build.txt.